### ISSUE TITLE
Pickling selectors not working

### ISSUE THREAD

**Issue by mgbckr** [flemmerich/pysubgroup#36]
Currently selectors are not picklable. 
```
import pickle
import pysubgroup as ps
a = ps.EqualitySelector("a", 1)
pickle.dump(a, open("tmp.p", "wb"))
pickle.load(open("tmp.p", "rb"))
```
This is due to the use of `__new__` without implementing `__getnewargs__` or `__getnewargs_ex__` (Python 3.6 and up) which provides the arguments used with `__new__` when unpickling ([source](https://docs.python.org/3/library/pickle.html#object.__getnewargs_ex__)).

**Comment by mgbckr** [comment]
Proposed fix (should have been a branch, sorry): #24f72a1202d79c7cc5b58d5624f09d392d59c8f1

**Comment by mgbckr** [comment]
Agreed.
